# Initialization and package preparation

In [2]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import pickle

%matplotlib inline
%load_ext autoreload
%autoreload 2

# Data load and preprocessing

In [15]:
import data_util
import data_preprocess
from sklearn.model_selection import train_test_split
from datetime import datetime
import numpy as np 

rso = np.random.RandomState(66)

train_data = data_util.load_train_data()
test_data= data_util.load_test_data()

lv1_pre = data_preprocess.preprocess_cell()

X_train = train_data.drop(['id','target'],axis=1)
y_train = train_data['target']
y_train

X_train, X_train_test, y_train, y_train_test = train_test_split(X_train,y_train,test_size =0.1 ,random_state=rso)
X_train, X_val, y_train, y_val = train_test_split(X_train,y_train, test_size =0.2, random_state=rso)

X_train, y_train = lv1_pre.process(X_train, y=y_train, rso =rso)
X_val = lv1_pre.process(X_val,test=True,rso =rso)
X_train_test = lv1_pre.process(X_train_test,test=True,rso =rso)
X_dev, y_dev = X_train[:10000,:], y_train[:10000]

# Load back neural network and give it a try

In [7]:
with open('search_lr_wd.pkl', 'rb') as f:
    neural_outcome = pickle.load(f)

In [22]:
import torch 
from torch.autograd import Variable
dtype = torch.FloatTensor


nn_model = neural_outcome['best_net']
nn_model.net.eval()
# X_train_test_tensor = torch.from_numpy(X_train_test.values)
# X_train_test_tensor= X_train_test_tensor.type(dtype)
# out = nn_model.test(X_train_test_tensor)
# nn_model.check_auc(out,y_train_test)


idx = test_data['id']
test_data_mutated = test_data.drop(['id'], axis=1)
test_data_mutated = naive_pre.dtype_change(test_data_mutated)
test_data_mutated = naive_pre.scale(test_data_mutated, test=True)
test_data_tensor = torch.from_numpy(test_data_mutated.values)
test_data_tensor = test_data_tensor.type(dtype)
out = nn_model.test(test_data_tensor)

In [44]:
result= np.hstack((idx.values.reshape(892816,1), out.data.numpy()[:,1].reshape(892816,1)))
print(result.shape)
result[:,0] = result[:,0].astype(np.int32)
np.savetxt('submission.csv',result, fmt=['%d', '%.9f'] , delimiter= ',', header= 'id, target')

(892816, 2)


In [ ]:
# ! cat submission.csv 10
#! sudo pip3 install lightgbm

# Try to use Lightgbm here.

In [2]:
! svn export https://github.com/Microsoft/LightGBM/tree/master/examples/python-guide

/bin/sh: 1: svn: not found


In [79]:
import data_util
import data_preprocess
from sklearn.model_selection import train_test_split
from datetime import datetime
import numpy as np 

rso = np.random.RandomState(66)

train_data = data_util.load_train_data()
test_data= data_util.load_test_data()

lv1_pre = data_preprocess.preprocess_cell()

X_train = train_data.drop(['id','target'],axis=1)
y_train = train_data['target']
y_train

X_train, X_train_test, y_train, y_train_test = train_test_split(X_train,y_train,test_size =0.1 ,random_state=rso)
X_train, X_val, y_train, y_val = train_test_split(X_train,y_train, test_size =0.2, random_state=rso)

X_train, y_train, col = lv1_pre.process(X_train, y=y_train, rso =rso)
X_val = lv1_pre.process(X_val,test=True,rso =rso)
X_train_test = lv1_pre.process(X_train_test,test=True,rso =rso)
X_dev, y_dev = X_train[:10000,:], y_train[:10000]

/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/sklearn/utils/deprecation.py:77: DeprecationWarning: Function _ratio_float is deprecated; Use a float for 'ratio' is deprecated from version 0.2. The support will be removed in 0.4. Use a dict, str, or a callable instead.
  warnings.warn(msg, category=DeprecationWarning)


In [ ]:

train_data11 = data_util.load_train_data()
train_data11.head(10)
# for col in train_data11.columns:
#     if 'bin' in col or 'cat' in col:
#         train_data11[col] = train_data11[col].astype('category')
# data = pd.get_dummies(train_data11)
# data.columns

uselessones = ['ps_ind_14',
             'ps_ind_10_bin',
             'ps_ind_11_bin',
             'ps_ind_12_bin', 
             'ps_ind_13_bin',
             'ps_ind_18_bin',
             'ps_car_10_cat',
             'ps_calc_15_bin',
             'ps_calc_20_bin'
        ]
print(train_data11.columns)
train_data11 = train_data11.drop(uselessones, axis =1)
print('hhhhh')
print(train_data11.columns)

# If increase data will boost my performance??
Well, not really, at least insignificantly.

In [27]:
import data_util
import data_preprocess
from sklearn.model_selection import train_test_split
from datetime import datetime
import numpy as np 
from sklearn import metrics
import lightgbm as lgb



rso = np.random.RandomState(66)

train_data = data_util.load_train_data()
lv1_pre = data_preprocess.preprocess_cell()
X_train = train_data.drop(['id','target'],axis=1)
y_train = train_data['target']

X_train, X_train_test, y_train, y_train_test = train_test_split(X_train,y_train,test_size =0.1 ,random_state=rso)
X_train, X_val, y_train, y_val = train_test_split(X_train,y_train, test_size =0.2, random_state=rso)

X_train, y_train, col = lv1_pre.process(X_train, y=y_train, rso =rso)
X_val = lv1_pre.process(X_val,test=True,rso =rso)
X_train_test = lv1_pre.process(X_train_test,test=True,rso =rso)

auc_dict={}

lgb_train = lgb.Dataset(X_train, y_train, free_raw_data=False)
lgb_val = lgb.Dataset(X_val, y_val, reference=lgb_train, free_raw_data=False)

#seems like lambda_l2 equals 0.0342
lgb1_params = {
    'boosting_type':'gbdt',
    'objective':'binary',
    'metric': 'binary_logloss',
    'lambda_l2':0.0342,
    'learning_rate': 0.09, 
    'feature_fraction': 0.9,   #
    'bagging_fraction': 0.8,   #
    'bagging_freq': 10,
    'verbose': 1,
    'verbose_eval':False
}
for i in range(1):
    
    #seems like the best combination is learning rate:0.1, Weight decay:0.05
    lr = 10**np.random.uniform(-2,0)
    #weight_decay = 10** np.random.uniform(-4,2)
    weight_decay = 10000000
    lgb1_params['learning_rate'] = lr
    lgb1_params['lambda_l2'] = weight_decay
    lgb1 = lgb.train(lgb1_params,
            lgb_train,
            num_boost_round =2500,  #Number of trees it will generate  
            valid_sets= lgb_val,
        early_stopping_rounds = 5
            )

    y_pred1 = lgb1.predict(X_train_test, lgb1.best_iteration)
    score = metrics.roc_auc_score(y_train_test,y_pred1)
    describe = '{}: We reach {} with lr {}, wd {} at the {} round!'.format(i+1, score,lr,weight_decay, lgb1.best_iteration)
    print(describe)
    auc_dict[describe] = score

[1]	valid_0's binary_logloss: 0.693147
Training until validation scores don't improve for 5 rounds.
[2]	valid_0's binary_logloss: 0.693147
[3]	valid_0's binary_logloss: 0.693147
[4]	valid_0's binary_logloss: 0.693147
[5]	valid_0's binary_logloss: 0.693147
[6]	valid_0's binary_logloss: 0.693147
Early stopping, best iteration is:
[1]	valid_0's binary_logloss: 0.693147
1: We reach 0.5 with lr 0.16928286101010034, wd 10000000 at the 1 round!


In [45]:
for i in sorted(auc_dict, key = lambda x: auc_dict[x], reverse=True):
    print(i)

71: We reach 0.6425146075906395 with lr 0.10090459507003365, wd 0.03426226856475561 at the 102 round!
148: We reach 0.6424326925153181 with lr 0.10745944670147414, wd 0.0003203867371549406 at the 83 round!
65: We reach 0.6421573746817226 with lr 0.04258541643753987, wd 0.21953639465793823 at the 237 round!
35: We reach 0.6421277053525621 with lr 0.013641124580520884, wd 0.6284206847642912 at the 760 round!
93: We reach 0.6421208784246081 with lr 0.014135085650213666, wd 9.483519863230152 at the 640 round!
31: We reach 0.6421177994559204 with lr 0.04477025632952884, wd 0.0010087862148837454 at the 218 round!
116: We reach 0.6419487947034517 with lr 0.01697460315960794, wd 0.0005371775659281679 at the 560 round!
36: We reach 0.6418749961764769 with lr 0.0140269328712022, wd 0.035941098081195236 at the 652 round!
7: We reach 0.6416946508246797 with lr 0.012987316213487771, wd 0.7248514625482201 at the 640 round!
42: We reach 0.6416197641804985 with lr 0.01567690910094347, wd 4.11173258472

In [3]:
import data_util
import data_preprocess
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import LabelEncoder
from datetime import datetime
import numpy as np 

rso = np.random.RandomState(66)

train_data = data_util.load_train_data()
test_data= data_util.load_test_data()

lv1_pre = data_preprocess.preprocess_cell()

X_train = train_data.drop(['id','target'],axis=1)
y_train = train_data['target']
X_test = test_data.drop(['id'], axis=1)

# combs = [
#     ('ps_reg_01', 'ps_car_02_cat'),  
#     ('ps_reg_01', 'ps_car_04_cat'),
# ]

# #'ps_reg_01_plus_ps_car_02_cat'
# #'ps_reg_01_plus_ps_car_04_cat'
# for n_c, (f1, f2) in enumerate(combs):
#     name1 = f1 + "_plus_" + f2

#     X_train[name1] = X_train[f1].apply(lambda x: str(x)) + "_" + X_train[f2].apply(lambda x: str(x))
#     X_test[name1] = X_test[f1].apply(lambda x: str(x)) + "_" + X_test[f2].apply(lambda x: str(x)) 
#     lbl = LabelEncoder()
#     lbl.fit(list(X_train[name1].values) + list(X_test[name1].values))
#     X_train[name1] = lbl.transform(list(X_train[name1].values))
#     X_test[name1] = lbl.transform(list(X_test[name1].values))


X_train, y_train, col = lv1_pre.process(X_train, y=y_train, rso =rso, oversample= None)
X_test = lv1_pre.process(X_test,test=True,rso =rso)

result = pd.DataFrame()
result['id'] = test_data['id']

In [ ]:
from Stacker import Ensemble

tic= datetime.now()
stack_model = Ensemble(n_splits=5)
y_pred= stack_model.fit_predict(X_train,y_train, X_test, META = 'nn')
toc = datetime.now()
print('This is round you consume {} time to run this model.'.format(toc-tic))

result['target'] = y_pred
result.to_csv('submission11.csv', index=False)

[1]	valid_0's binary_logloss: 0.651205
Training until validation scores don't improve for 5 rounds.
[2]	valid_0's binary_logloss: 0.613263
[3]	valid_0's binary_logloss: 0.578807
[4]	valid_0's binary_logloss: 0.547396
[5]	valid_0's binary_logloss: 0.51868
[6]	valid_0's binary_logloss: 0.492351
[7]	valid_0's binary_logloss: 0.468154
[8]	valid_0's binary_logloss: 0.445878
[9]	valid_0's binary_logloss: 0.425329
[10]	valid_0's binary_logloss: 0.406339
[11]	valid_0's binary_logloss: 0.388766
[12]	valid_0's binary_logloss: 0.372485
[13]	valid_0's binary_logloss: 0.357377
[14]	valid_0's binary_logloss: 0.343355
[15]	valid_0's binary_logloss: 0.330322
[16]	valid_0's binary_logloss: 0.318198
[17]	valid_0's binary_logloss: 0.306918
[18]	valid_0's binary_logloss: 0.296413
[19]	valid_0's binary_logloss: 0.286623
[20]	valid_0's binary_logloss: 0.277504
[21]	valid_0's binary_logloss: 0.268994
[22]	valid_0's binary_logloss: 0.261055
[23]	valid_0's binary_logloss: 0.253646
[24]	valid_0's binary_logloss

[48]	valid_0's binary_logloss: 0.168465
[49]	valid_0's binary_logloss: 0.167275
[50]	valid_0's binary_logloss: 0.16617
[51]	valid_0's binary_logloss: 0.165141
[52]	valid_0's binary_logloss: 0.164187
[53]	valid_0's binary_logloss: 0.16331
[54]	valid_0's binary_logloss: 0.162496
[55]	valid_0's binary_logloss: 0.16173
[56]	valid_0's binary_logloss: 0.161021
[57]	valid_0's binary_logloss: 0.160373
[58]	valid_0's binary_logloss: 0.159761
[59]	valid_0's binary_logloss: 0.159195
[60]	valid_0's binary_logloss: 0.158683
[61]	valid_0's binary_logloss: 0.158208
[62]	valid_0's binary_logloss: 0.157765
[63]	valid_0's binary_logloss: 0.157352
[64]	valid_0's binary_logloss: 0.156966
[65]	valid_0's binary_logloss: 0.156611
[66]	valid_0's binary_logloss: 0.156286
[67]	valid_0's binary_logloss: 0.155982
[68]	valid_0's binary_logloss: 0.155704
[69]	valid_0's binary_logloss: 0.155441
[70]	valid_0's binary_logloss: 0.155206
[71]	valid_0's binary_logloss: 0.154989
[72]	valid_0's binary_logloss: 0.154794
[73

[75]	valid_0's binary_logloss: 0.154297
[76]	valid_0's binary_logloss: 0.154153
[77]	valid_0's binary_logloss: 0.154015
[78]	valid_0's binary_logloss: 0.153885
[79]	valid_0's binary_logloss: 0.153766
[80]	valid_0's binary_logloss: 0.15366
[81]	valid_0's binary_logloss: 0.153559
[82]	valid_0's binary_logloss: 0.153468
[83]	valid_0's binary_logloss: 0.153384
[84]	valid_0's binary_logloss: 0.15331
[85]	valid_0's binary_logloss: 0.153239
[86]	valid_0's binary_logloss: 0.153167
[87]	valid_0's binary_logloss: 0.153096
[88]	valid_0's binary_logloss: 0.153041
[89]	valid_0's binary_logloss: 0.152986
[90]	valid_0's binary_logloss: 0.15294
[91]	valid_0's binary_logloss: 0.152883
[92]	valid_0's binary_logloss: 0.152841
[93]	valid_0's binary_logloss: 0.152794
[94]	valid_0's binary_logloss: 0.152747
[95]	valid_0's binary_logloss: 0.152707
[96]	valid_0's binary_logloss: 0.152674
[97]	valid_0's binary_logloss: 0.152643
[98]	valid_0's binary_logloss: 0.152605
[99]	valid_0's binary_logloss: 0.152567
[10

[130]	valid_0's binary_logloss: 0.151903
[131]	valid_0's binary_logloss: 0.151902
[132]	valid_0's binary_logloss: 0.151893
[133]	valid_0's binary_logloss: 0.151884
[134]	valid_0's binary_logloss: 0.151879
[135]	valid_0's binary_logloss: 0.151874
[136]	valid_0's binary_logloss: 0.151871
[137]	valid_0's binary_logloss: 0.151856
[138]	valid_0's binary_logloss: 0.151855
[139]	valid_0's binary_logloss: 0.151857
[140]	valid_0's binary_logloss: 0.151855
[141]	valid_0's binary_logloss: 0.151839
[142]	valid_0's binary_logloss: 0.151822
[143]	valid_0's binary_logloss: 0.151818
[144]	valid_0's binary_logloss: 0.151805
[145]	valid_0's binary_logloss: 0.151797
[146]	valid_0's binary_logloss: 0.151792
[147]	valid_0's binary_logloss: 0.151781
[148]	valid_0's binary_logloss: 0.151779
[149]	valid_0's binary_logloss: 0.151776
[150]	valid_0's binary_logloss: 0.151781
[151]	valid_0's binary_logloss: 0.151779
[152]	valid_0's binary_logloss: 0.15177
[153]	valid_0's binary_logloss: 0.151763
[154]	valid_0's b

[164]	valid_0's binary_logloss: 0.151943
[165]	valid_0's binary_logloss: 0.151937
[166]	valid_0's binary_logloss: 0.151939
[167]	valid_0's binary_logloss: 0.151933
[168]	valid_0's binary_logloss: 0.151932
[169]	valid_0's binary_logloss: 0.151928
[170]	valid_0's binary_logloss: 0.151921
[171]	valid_0's binary_logloss: 0.151917
[172]	valid_0's binary_logloss: 0.151914
[173]	valid_0's binary_logloss: 0.151911
[174]	valid_0's binary_logloss: 0.151902
[175]	valid_0's binary_logloss: 0.151898
[176]	valid_0's binary_logloss: 0.151896
[177]	valid_0's binary_logloss: 0.151892
[178]	valid_0's binary_logloss: 0.15189
[179]	valid_0's binary_logloss: 0.151892
[180]	valid_0's binary_logloss: 0.151887
[181]	valid_0's binary_logloss: 0.151885
[182]	valid_0's binary_logloss: 0.151891
[183]	valid_0's binary_logloss: 0.15189
[184]	valid_0's binary_logloss: 0.151884
[185]	valid_0's binary_logloss: 0.151883
[186]	valid_0's binary_logloss: 0.151883
[187]	valid_0's binary_logloss: 0.151884
[188]	valid_0's bi

[8]	valid_0's binary_logloss: 0.31499
[9]	valid_0's binary_logloss: 0.293408
[10]	valid_0's binary_logloss: 0.274701
[11]	valid_0's binary_logloss: 0.258467
[12]	valid_0's binary_logloss: 0.244356
[13]	valid_0's binary_logloss: 0.232079
[14]	valid_0's binary_logloss: 0.221392
[15]	valid_0's binary_logloss: 0.21209
[16]	valid_0's binary_logloss: 0.203992
[17]	valid_0's binary_logloss: 0.196951
[18]	valid_0's binary_logloss: 0.190828
[19]	valid_0's binary_logloss: 0.185517
[20]	valid_0's binary_logloss: 0.180907
[21]	valid_0's binary_logloss: 0.176919
[22]	valid_0's binary_logloss: 0.173461
[23]	valid_0's binary_logloss: 0.170481
[24]	valid_0's binary_logloss: 0.167912
[25]	valid_0's binary_logloss: 0.165675
[26]	valid_0's binary_logloss: 0.163781
[27]	valid_0's binary_logloss: 0.162153
[28]	valid_0's binary_logloss: 0.160728
[29]	valid_0's binary_logloss: 0.159521
[30]	valid_0's binary_logloss: 0.158474
[31]	valid_0's binary_logloss: 0.157598
[32]	valid_0's binary_logloss: 0.156812
[33]

[49]	valid_0's binary_logloss: 0.152584
[50]	valid_0's binary_logloss: 0.152536
[51]	valid_0's binary_logloss: 0.152491
[52]	valid_0's binary_logloss: 0.152459
[53]	valid_0's binary_logloss: 0.152415
[54]	valid_0's binary_logloss: 0.152379
[55]	valid_0's binary_logloss: 0.152341
[56]	valid_0's binary_logloss: 0.152303
[57]	valid_0's binary_logloss: 0.152284
[58]	valid_0's binary_logloss: 0.152257
[59]	valid_0's binary_logloss: 0.152239
[60]	valid_0's binary_logloss: 0.152211
[61]	valid_0's binary_logloss: 0.152194
[62]	valid_0's binary_logloss: 0.152155
[63]	valid_0's binary_logloss: 0.152128
[64]	valid_0's binary_logloss: 0.152098
[65]	valid_0's binary_logloss: 0.152073
[66]	valid_0's binary_logloss: 0.152046
[67]	valid_0's binary_logloss: 0.152046
[68]	valid_0's binary_logloss: 0.152042
[69]	valid_0's binary_logloss: 0.152049
[70]	valid_0's binary_logloss: 0.152029
[71]	valid_0's binary_logloss: 0.152021
[72]	valid_0's binary_logloss: 0.151999
[73]	valid_0's binary_logloss: 0.151995


[157]	valid_0's binary_logloss: 0.15248
[158]	valid_0's binary_logloss: 0.15245
[159]	valid_0's binary_logloss: 0.152425
[160]	valid_0's binary_logloss: 0.152406
[161]	valid_0's binary_logloss: 0.152384
[162]	valid_0's binary_logloss: 0.152367
[163]	valid_0's binary_logloss: 0.152345
[164]	valid_0's binary_logloss: 0.152322
[165]	valid_0's binary_logloss: 0.152305
[166]	valid_0's binary_logloss: 0.152286
[167]	valid_0's binary_logloss: 0.152265
[168]	valid_0's binary_logloss: 0.152248
[169]	valid_0's binary_logloss: 0.152232
[170]	valid_0's binary_logloss: 0.152217
[171]	valid_0's binary_logloss: 0.152203
[172]	valid_0's binary_logloss: 0.152191
[173]	valid_0's binary_logloss: 0.152174
[174]	valid_0's binary_logloss: 0.152159
[175]	valid_0's binary_logloss: 0.152144
[176]	valid_0's binary_logloss: 0.152129
[177]	valid_0's binary_logloss: 0.152115
[178]	valid_0's binary_logloss: 0.152103
[179]	valid_0's binary_logloss: 0.152086
[180]	valid_0's binary_logloss: 0.15208
[181]	valid_0's bin

[37]	valid_0's binary_logloss: 0.260522
[38]	valid_0's binary_logloss: 0.256046
[39]	valid_0's binary_logloss: 0.251757
[40]	valid_0's binary_logloss: 0.247648
[41]	valid_0's binary_logloss: 0.243704
[42]	valid_0's binary_logloss: 0.239921
[43]	valid_0's binary_logloss: 0.236282
[44]	valid_0's binary_logloss: 0.232791
[45]	valid_0's binary_logloss: 0.229443
[46]	valid_0's binary_logloss: 0.226228
[47]	valid_0's binary_logloss: 0.223142
[48]	valid_0's binary_logloss: 0.22018
[49]	valid_0's binary_logloss: 0.217342
[50]	valid_0's binary_logloss: 0.214615
[51]	valid_0's binary_logloss: 0.212002
[52]	valid_0's binary_logloss: 0.209497
[53]	valid_0's binary_logloss: 0.207092
[54]	valid_0's binary_logloss: 0.204788
[55]	valid_0's binary_logloss: 0.202564
[56]	valid_0's binary_logloss: 0.200431
[57]	valid_0's binary_logloss: 0.198382
[58]	valid_0's binary_logloss: 0.196419
[59]	valid_0's binary_logloss: 0.194535
[60]	valid_0's binary_logloss: 0.192724
[61]	valid_0's binary_logloss: 0.191002
[

[241]	valid_0's binary_logloss: 0.152223
[242]	valid_0's binary_logloss: 0.152221
[243]	valid_0's binary_logloss: 0.152215
[244]	valid_0's binary_logloss: 0.152215
[245]	valid_0's binary_logloss: 0.152212
[246]	valid_0's binary_logloss: 0.152211
[247]	valid_0's binary_logloss: 0.152209
[248]	valid_0's binary_logloss: 0.152207
[249]	valid_0's binary_logloss: 0.152202
[250]	valid_0's binary_logloss: 0.152202
[251]	valid_0's binary_logloss: 0.1522
[252]	valid_0's binary_logloss: 0.152198
[253]	valid_0's binary_logloss: 0.152192
[254]	valid_0's binary_logloss: 0.152192
[255]	valid_0's binary_logloss: 0.152191
[256]	valid_0's binary_logloss: 0.152192
[257]	valid_0's binary_logloss: 0.152189
[258]	valid_0's binary_logloss: 0.152185
[259]	valid_0's binary_logloss: 0.152182
[260]	valid_0's binary_logloss: 0.152185
[261]	valid_0's binary_logloss: 0.152183
[262]	valid_0's binary_logloss: 0.152183
[263]	valid_0's binary_logloss: 0.152181
[264]	valid_0's binary_logloss: 0.152178
[265]	valid_0's bi

[157]	valid_0's binary_logloss: 0.152737
[158]	valid_0's binary_logloss: 0.152717
[159]	valid_0's binary_logloss: 0.152697
[160]	valid_0's binary_logloss: 0.152676
[161]	valid_0's binary_logloss: 0.152657
[162]	valid_0's binary_logloss: 0.152644
[163]	valid_0's binary_logloss: 0.152621
[164]	valid_0's binary_logloss: 0.152602
[165]	valid_0's binary_logloss: 0.152582
[166]	valid_0's binary_logloss: 0.152562
[167]	valid_0's binary_logloss: 0.152545
[168]	valid_0's binary_logloss: 0.152528
[169]	valid_0's binary_logloss: 0.152512
[170]	valid_0's binary_logloss: 0.152489
[171]	valid_0's binary_logloss: 0.152472
[172]	valid_0's binary_logloss: 0.152455
[173]	valid_0's binary_logloss: 0.152439
[174]	valid_0's binary_logloss: 0.152429
[175]	valid_0's binary_logloss: 0.152423
[176]	valid_0's binary_logloss: 0.152414
[177]	valid_0's binary_logloss: 0.152404
[178]	valid_0's binary_logloss: 0.152396
[179]	valid_0's binary_logloss: 0.152382
[180]	valid_0's binary_logloss: 0.152375
[181]	valid_0's 

[85]	valid_0's binary_logloss: 0.165369
[86]	valid_0's binary_logloss: 0.164772
[87]	valid_0's binary_logloss: 0.164195
[88]	valid_0's binary_logloss: 0.163656
[89]	valid_0's binary_logloss: 0.163135
[90]	valid_0's binary_logloss: 0.162637
[91]	valid_0's binary_logloss: 0.162164
[92]	valid_0's binary_logloss: 0.161715
[93]	valid_0's binary_logloss: 0.161278
[94]	valid_0's binary_logloss: 0.160868
[95]	valid_0's binary_logloss: 0.16047
[96]	valid_0's binary_logloss: 0.160101
[97]	valid_0's binary_logloss: 0.159739
[98]	valid_0's binary_logloss: 0.159392
[99]	valid_0's binary_logloss: 0.15906
[100]	valid_0's binary_logloss: 0.158745
[101]	valid_0's binary_logloss: 0.158438
[102]	valid_0's binary_logloss: 0.158143
[103]	valid_0's binary_logloss: 0.15787
[104]	valid_0's binary_logloss: 0.157614
[105]	valid_0's binary_logloss: 0.157368
[106]	valid_0's binary_logloss: 0.157134
[107]	valid_0's binary_logloss: 0.156905
[108]	valid_0's binary_logloss: 0.156685
[109]	valid_0's binary_logloss: 0.

[1]	valid_0's binary_logloss: 0.667707
Training until validation scores don't improve for 5 rounds.
[2]	valid_0's binary_logloss: 0.643752
[3]	valid_0's binary_logloss: 0.621155
[4]	valid_0's binary_logloss: 0.599837
[5]	valid_0's binary_logloss: 0.579675
[6]	valid_0's binary_logloss: 0.560585
[7]	valid_0's binary_logloss: 0.542504
[8]	valid_0's binary_logloss: 0.525351
[9]	valid_0's binary_logloss: 0.509068
[10]	valid_0's binary_logloss: 0.49359
[11]	valid_0's binary_logloss: 0.478869
[12]	valid_0's binary_logloss: 0.464857
[13]	valid_0's binary_logloss: 0.451513
[14]	valid_0's binary_logloss: 0.438797
[15]	valid_0's binary_logloss: 0.426672
[16]	valid_0's binary_logloss: 0.415095
[17]	valid_0's binary_logloss: 0.404044
[18]	valid_0's binary_logloss: 0.393491
[19]	valid_0's binary_logloss: 0.383417
[20]	valid_0's binary_logloss: 0.373791
[21]	valid_0's binary_logloss: 0.364586
[22]	valid_0's binary_logloss: 0.35577
[23]	valid_0's binary_logloss: 0.347334
[24]	valid_0's binary_logloss:

[205]	valid_0's binary_logloss: 0.152203
[206]	valid_0's binary_logloss: 0.152202
[207]	valid_0's binary_logloss: 0.152193
[208]	valid_0's binary_logloss: 0.15219
[209]	valid_0's binary_logloss: 0.152189
[210]	valid_0's binary_logloss: 0.152186
[211]	valid_0's binary_logloss: 0.152178
[212]	valid_0's binary_logloss: 0.152173
[213]	valid_0's binary_logloss: 0.152162
[214]	valid_0's binary_logloss: 0.152153
[215]	valid_0's binary_logloss: 0.152156
[216]	valid_0's binary_logloss: 0.152149
[217]	valid_0's binary_logloss: 0.152149
[218]	valid_0's binary_logloss: 0.152144
[219]	valid_0's binary_logloss: 0.152138
[220]	valid_0's binary_logloss: 0.152133
[221]	valid_0's binary_logloss: 0.152133
[222]	valid_0's binary_logloss: 0.152127
[223]	valid_0's binary_logloss: 0.152124
[224]	valid_0's binary_logloss: 0.152118
[225]	valid_0's binary_logloss: 0.152115
[226]	valid_0's binary_logloss: 0.152116
[227]	valid_0's binary_logloss: 0.152114
[228]	valid_0's binary_logloss: 0.152111
[229]	valid_0's b

Epoch 1: iteration 700, the loss is [ 0.1631472]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6353695461519324, auc for val: 0.6356473717593167
--------------------------------------------------------------
Epoch 1: iteration 800, the loss is [ 0.21186188]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6348519210896066, auc for val: 0.6351094582420296
--------------------------------------------------------------
Epoch 1: iteration 900, the loss is [ 0.17485549]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6328981801020968, auc for val: 0.633874896115224
--------------------------------------------------------------
Epoch 1: iteration 1000, the loss is [ 0.1812806]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6344001573935877, auc for val: 0.6347850668597186
--------------------------------------------------------------
Epoch 1: iteration 1100, the loss is [ 0.17109

Epoch 4: iteration 600, the loss is [ 0.17123534]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6405515372169992, auc for val: 0.6423154587868063
--------------------------------------------------------------
Epoch 4: iteration 700, the loss is [ 0.14114846]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6401530472996687, auc for val: 0.6418341209364272
--------------------------------------------------------------
Epoch 4: iteration 800, the loss is [ 0.17291428]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6403812612473724, auc for val: 0.6423939772666689
--------------------------------------------------------------
Epoch 4: iteration 900, the loss is [ 0.13747206]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6409457073357301, auc for val: 0.6427124216503789
--------------------------------------------------------------
Epoch 4: iteration 1000, the loss is [ 0.162

Epoch 7: iteration 500, the loss is [ 0.15307431]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.64088473489334, auc for val: 0.6426355871637007
--------------------------------------------------------------
Epoch 7: iteration 600, the loss is [ 0.14458653]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6409408652696099, auc for val: 0.6427008783083492
--------------------------------------------------------------
Epoch 7: iteration 700, the loss is [ 0.19057295]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6409441255836822, auc for val: 0.6427465751813357
--------------------------------------------------------------
Epoch 7: iteration 800, the loss is [ 0.15651868]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6408814534008294, auc for val: 0.642540189161894
--------------------------------------------------------------
Epoch 7: iteration 900, the loss is [ 0.1325729

Epoch 10: iteration 400, the loss is [ 0.18062015]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6405471488999229, auc for val: 0.6422062339425558
--------------------------------------------------------------
Epoch 10: iteration 500, the loss is [ 0.2074195]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6407434591579804, auc for val: 0.6423324740462363
--------------------------------------------------------------
Epoch 10: iteration 600, the loss is [ 0.15216368]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6406104202276254, auc for val: 0.6422619768034308
--------------------------------------------------------------
Epoch 10: iteration 700, the loss is [ 0.18350489]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6405069468792973, auc for val: 0.6422693151112621
--------------------------------------------------------------
Epoch 10: iteration 800, the loss is [ 0.

Epoch 13: iteration 300, the loss is [ 0.16375928]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6405888305981503, auc for val: 0.6424322439262347
--------------------------------------------------------------
Epoch 13: iteration 400, the loss is [ 0.10325548]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6405790920128098, auc for val: 0.6424467452143112
--------------------------------------------------------------
Epoch 13: iteration 500, the loss is [ 0.13005342]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6406352089265053, auc for val: 0.6422670712009484
--------------------------------------------------------------
Epoch 13: iteration 600, the loss is [ 0.16938886]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6404639038937512, auc for val: 0.6422423966710902
--------------------------------------------------------------
Epoch 13: iteration 700, the loss is [ 0

Epoch 16: iteration 200, the loss is [ 0.16945383]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6404929058962375, auc for val: 0.6422084806807339
--------------------------------------------------------------
Epoch 16: iteration 300, the loss is [ 0.18169078]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6406833878641657, auc for val: 0.6423753529524582
--------------------------------------------------------------
Epoch 16: iteration 400, the loss is [ 0.17143849]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6403100451118803, auc for val: 0.642044757235896
--------------------------------------------------------------
Epoch 16: iteration 500, the loss is [ 0.13725767]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6406577419006849, auc for val: 0.6423744918677821
--------------------------------------------------------------
Epoch 16: iteration 600, the loss is [ 0.

Epoch 19: iteration 100, the loss is [ 0.14438835]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.640269232252049, auc for val: 0.6419210636240082
--------------------------------------------------------------
Epoch 19: iteration 200, the loss is [ 0.13145542]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6402853019044465, auc for val: 0.6419544239390318
--------------------------------------------------------------
Epoch 19: iteration 300, the loss is [ 0.17170766]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6406253638059849, auc for val: 0.6423879327067501
--------------------------------------------------------------
Epoch 19: iteration 400, the loss is [ 0.16417311]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6407717070151686, auc for val: 0.6424787736052543
--------------------------------------------------------------
Epoch 19: iteration 500, the loss is [ 0.

Epoch 22: iteration 0, the loss is [ 0.15243264]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.640661952230469, auc for val: 0.6422377391786054
--------------------------------------------------------------
Epoch 22: iteration 100, the loss is [ 0.13250232]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6402834993661755, auc for val: 0.6419124612608397
--------------------------------------------------------------
Epoch 22: iteration 200, the loss is [ 0.19594081]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6405448670136227, auc for val: 0.6421182194944091
--------------------------------------------------------------
Epoch 22: iteration 300, the loss is [ 0.08763209]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6405016437501374, auc for val: 0.6420507169598861
--------------------------------------------------------------
Epoch 22: iteration 400, the loss is [ 0.14

Epoch 24: iteration 1100, the loss is [ 0.19413586]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6405848308471451, auc for val: 0.6422364836068608
--------------------------------------------------------------
Epoch 25: iteration 0, the loss is [ 0.1414497]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6406086259678331, auc for val: 0.6421911458726391
--------------------------------------------------------------
Epoch 25: iteration 100, the loss is [ 0.10052525]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6403401804220265, auc for val: 0.6419895191487968
--------------------------------------------------------------
Epoch 25: iteration 200, the loss is [ 0.14733085]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6404272662404497, auc for val: 0.6421052707038273
--------------------------------------------------------------
Epoch 25: iteration 300, the loss is [ 0.1

Epoch 27: iteration 1000, the loss is [ 0.17144346]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6404279911297647, auc for val: 0.6420920730611882
--------------------------------------------------------------
Epoch 27: iteration 1100, the loss is [ 0.1321985]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.640524874125868, auc for val: 0.6422107542836226
--------------------------------------------------------------
Epoch 28: iteration 0, the loss is [ 0.12761624]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6404358499780107, auc for val: 0.6421321071359347
--------------------------------------------------------------
Epoch 28: iteration 100, the loss is [ 0.17987025]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6406665332432022, auc for val: 0.642301303912105
--------------------------------------------------------------
Epoch 28: iteration 200, the loss is [ 0.19

Epoch 30: iteration 900, the loss is [ 0.20856793]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6406825232051955, auc for val: 0.6423749146334932
--------------------------------------------------------------
Epoch 30: iteration 1000, the loss is [ 0.18689081]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6404344204133361, auc for val: 0.6420993152216337
--------------------------------------------------------------
Epoch 30: iteration 1100, the loss is [ 0.17332338]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6405643449880394, auc for val: 0.6421691366048738
--------------------------------------------------------------
Epoch 31: iteration 0, the loss is [ 0.11813226]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6404628985810167, auc for val: 0.6421289173050163
--------------------------------------------------------------
Epoch 31: iteration 100, the loss is [ 0

Epoch 33: iteration 800, the loss is [ 0.16771905]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6404857674363859, auc for val: 0.6422079349029259
--------------------------------------------------------------
Epoch 33: iteration 900, the loss is [ 0.15204683]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6403385481551839, auc for val: 0.6420223308581481
--------------------------------------------------------------
Epoch 33: iteration 1000, the loss is [ 0.14227487]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6402963917304082, auc for val: 0.6420334146722307
--------------------------------------------------------------
Epoch 33: iteration 1100, the loss is [ 0.14593346]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6404907655076242, auc for val: 0.642205268226901
--------------------------------------------------------------
Epoch 34: iteration 0, the loss is [ 0.

Epoch 36: iteration 700, the loss is [ 0.17838658]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6405180261361333, auc for val: 0.6422292697250582
--------------------------------------------------------------
Epoch 36: iteration 800, the loss is [ 0.15344155]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6404326382497674, auc for val: 0.6421328353109894
--------------------------------------------------------------
Epoch 36: iteration 900, the loss is [ 0.16628027]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6404834419474186, auc for val: 0.6422145945233277
--------------------------------------------------------------
Epoch 36: iteration 1000, the loss is [ 0.14365144]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6405375475926566, auc for val: 0.6422296981464981
--------------------------------------------------------------
Epoch 36: iteration 1100, the loss is [

Epoch 39: iteration 600, the loss is [ 0.18496211]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6405242494624068, auc for val: 0.6422188815655908
--------------------------------------------------------------
Epoch 39: iteration 700, the loss is [ 0.12952688]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6404762393092638, auc for val: 0.6421713508226127
--------------------------------------------------------------
Epoch 39: iteration 800, the loss is [ 0.14720708]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6404399359296801, auc for val: 0.6421200703315866
--------------------------------------------------------------
Epoch 39: iteration 900, the loss is [ 0.14904442]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6406923954116122, auc for val: 0.6423942741924193
--------------------------------------------------------------
Epoch 39: iteration 1000, the loss is [ 

Epoch 42: iteration 500, the loss is [ 0.17590813]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6404282517812836, auc for val: 0.6421311145555693
--------------------------------------------------------------
Epoch 42: iteration 600, the loss is [ 0.14297199]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6404432521435771, auc for val: 0.642123610817677
--------------------------------------------------------------
Epoch 42: iteration 700, the loss is [ 0.13285777]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.640432106739285, auc for val: 0.6421044944550798
--------------------------------------------------------------
Epoch 42: iteration 800, the loss is [ 0.1970634]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6404222973046342, auc for val: 0.642087655937168
--------------------------------------------------------------
Epoch 42: iteration 900, the loss is [ 0.119

Epoch 45: iteration 400, the loss is [ 0.15344939]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6404253287147516, auc for val: 0.6420918185534021
--------------------------------------------------------------
Epoch 45: iteration 500, the loss is [ 0.13301294]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6404589828605891, auc for val: 0.6421218391606996
--------------------------------------------------------------
Epoch 45: iteration 600, the loss is [ 0.16338055]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6405822841298998, auc for val: 0.6421761850566156
--------------------------------------------------------------
Epoch 45: iteration 700, the loss is [ 0.11523358]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6405304221131382, auc for val: 0.6422006149762126
--------------------------------------------------------------
Epoch 45: iteration 800, the loss is [ 0

Epoch 48: iteration 300, the loss is [ 0.18043232]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6404343002548344, auc for val: 0.6421146394182184
--------------------------------------------------------------
Epoch 48: iteration 400, the loss is [ 0.14553396]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6404829538386694, auc for val: 0.6422008397914236
--------------------------------------------------------------
Epoch 48: iteration 500, the loss is [ 0.14744569]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6404602337146093, auc for val: 0.64213747018056
--------------------------------------------------------------
Epoch 48: iteration 600, the loss is [ 0.15659545]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6404516974376471, auc for val: 0.642112924318527
--------------------------------------------------------------
Epoch 48: iteration 700, the loss is [ 0.13

Epoch 51: iteration 200, the loss is [ 0.18785766]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6404812112992755, auc for val: 0.6421860613726467
--------------------------------------------------------------
Epoch 51: iteration 300, the loss is [ 0.11964673]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6404608614322665, auc for val: 0.6421281650931152
--------------------------------------------------------------
Epoch 51: iteration 400, the loss is [ 0.16745371]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6404296834758582, auc for val: 0.6421090515583825
--------------------------------------------------------------
Epoch 51: iteration 500, the loss is [ 0.19081527]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6405286628962266, auc for val: 0.6421904035582632
--------------------------------------------------------------
Epoch 51: iteration 600, the loss is [ 0

Epoch 54: iteration 100, the loss is [ 0.13409941]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.640535021973476, auc for val: 0.6422022508957042
--------------------------------------------------------------
Epoch 54: iteration 200, the loss is [ 0.11104378]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6404396336446472, auc for val: 0.6421043431643404
--------------------------------------------------------------
Epoch 54: iteration 300, the loss is [ 0.17271379]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6404831237484168, auc for val: 0.6421777276565854
--------------------------------------------------------------
Epoch 54: iteration 400, the loss is [ 0.15544707]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6404619891539629, auc for val: 0.6421233195476552
--------------------------------------------------------------
Epoch 54: iteration 500, the loss is [ 0.

Epoch 57: iteration 0, the loss is [ 0.12593019]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6405431851160945, auc for val: 0.6422046291295715
--------------------------------------------------------------
Epoch 57: iteration 100, the loss is [ 0.13583386]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6405185449074032, auc for val: 0.6422381972926202
--------------------------------------------------------------
Epoch 57: iteration 200, the loss is [ 0.12651226]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6404821855476213, auc for val: 0.6421840083431725
--------------------------------------------------------------
Epoch 57: iteration 300, the loss is [ 0.17672005]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6404793435780313, auc for val: 0.6421925682883767
--------------------------------------------------------------
Epoch 57: iteration 400, the loss is [ 0.1

Epoch 59: iteration 1100, the loss is [ 0.18208]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.640635887601012, auc for val: 0.6422335468697962
--------------------------------------------------------------
Learning rate is 0.0006241303598307956. Weight decay is 0.0026706282114709093. dropout is 0.5
 Val aus is 0.6422335468697962. Train auc is 0.640635887601012
BLOWINGUP: Learning rate is 0.0006241303598307956. Weight decay is 0.0026706282114709093. dropout is 0.5
 Val aus is 0.6422335468697962. Train auc is 0.640635887601012
Epoch 0: iteration 0, the loss is [ 0.72375709]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.3618346059936167, auc for val: 0.36121835495787485
--------------------------------------------------------------
Epoch 0: iteration 100, the loss is [ 0.59839362]
  acc for train: 0.9634583980161691, acc for val: 0.9632
  auc for train: 0.36894363123183477, auc for val: 0.3688886809214649
-----------------------

Epoch 2: iteration 800, the loss is [ 0.18292734]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6408074564600411, auc for val: 0.6424712472444478
--------------------------------------------------------------
Epoch 2: iteration 900, the loss is [ 0.20819022]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6402649091983197, auc for val: 0.6423129009835564
--------------------------------------------------------------
Epoch 2: iteration 1000, the loss is [ 0.18566027]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6397002750357855, auc for val: 0.6411602578434217
--------------------------------------------------------------
Epoch 2: iteration 1100, the loss is [ 0.16648756]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6407774405850345, auc for val: 0.6425852186589079
--------------------------------------------------------------
Epoch 3: iteration 0, the loss is [ 0.1767

Epoch 5: iteration 700, the loss is [ 0.12625016]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6412495521789092, auc for val: 0.6426763126512662
--------------------------------------------------------------
Epoch 5: iteration 800, the loss is [ 0.19907188]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6413283024533857, auc for val: 0.6425937969852302
--------------------------------------------------------------
Epoch 5: iteration 900, the loss is [ 0.15510945]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6412682119904718, auc for val: 0.6424564037847921
--------------------------------------------------------------
Epoch 5: iteration 1000, the loss is [ 0.17080772]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6413354005255134, auc for val: 0.6426735653811089
--------------------------------------------------------------
Epoch 5: iteration 1100, the loss is [ 0.17

Epoch 8: iteration 600, the loss is [ 0.16101722]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6412264657522602, auc for val: 0.6430053473867405
--------------------------------------------------------------
Epoch 8: iteration 700, the loss is [ 0.14648403]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.641158570411959, auc for val: 0.6426527098819738
--------------------------------------------------------------
Epoch 8: iteration 800, the loss is [ 0.16347633]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.641181739703769, auc for val: 0.6426810917419155
--------------------------------------------------------------
Epoch 8: iteration 900, the loss is [ 0.14594373]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6412883534889753, auc for val: 0.6428222940755517
--------------------------------------------------------------
Epoch 8: iteration 1000, the loss is [ 0.15691

Epoch 11: iteration 500, the loss is [ 0.17340796]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6413679085039179, auc for val: 0.6428830252890332
--------------------------------------------------------------
Epoch 11: iteration 600, the loss is [ 0.13652141]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6413630739929144, auc for val: 0.6429244817795174
--------------------------------------------------------------
Epoch 11: iteration 700, the loss is [ 0.15047871]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6413600222482813, auc for val: 0.6431005799584864
--------------------------------------------------------------
Epoch 11: iteration 800, the loss is [ 0.14771162]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6412690706616102, auc for val: 0.643168021693927
--------------------------------------------------------------
Epoch 11: iteration 900, the loss is [ 0.

Epoch 14: iteration 400, the loss is [ 0.18012829]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6413629885156122, auc for val: 0.6430525925154255
--------------------------------------------------------------
Epoch 14: iteration 500, the loss is [ 0.19677559]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6413431786384985, auc for val: 0.6428461527665623
--------------------------------------------------------------
Epoch 14: iteration 600, the loss is [ 0.18527341]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6413755207258349, auc for val: 0.6428948669707456
--------------------------------------------------------------
Epoch 14: iteration 700, the loss is [ 0.16303727]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6413747213703984, auc for val: 0.6429328593274752
--------------------------------------------------------------
Epoch 14: iteration 800, the loss is [ 0

Epoch 17: iteration 300, the loss is [ 0.16728979]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6413570314670084, auc for val: 0.64297369651568
--------------------------------------------------------------
Epoch 17: iteration 400, the loss is [ 0.15872625]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6413737079801192, auc for val: 0.6430296641917661
--------------------------------------------------------------
Epoch 17: iteration 500, the loss is [ 0.14915104]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6413733823063743, auc for val: 0.6430367706147255
--------------------------------------------------------------
Epoch 17: iteration 600, the loss is [ 0.14776921]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6413693034275799, auc for val: 0.6430210901072402
--------------------------------------------------------------
Epoch 17: iteration 700, the loss is [ 0.1

Epoch 20: iteration 200, the loss is [ 0.16090731]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6413325211018349, auc for val: 0.6430941720180063
--------------------------------------------------------------
Epoch 20: iteration 300, the loss is [ 0.12473518]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6413668506268625, auc for val: 0.6431271802639256
--------------------------------------------------------------
Epoch 20: iteration 400, the loss is [ 0.15205887]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6413657236687152, auc for val: 0.6431702769156978
--------------------------------------------------------------
Epoch 20: iteration 500, the loss is [ 0.17975491]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6413571043256583, auc for val: 0.6431576688827632
--------------------------------------------------------------
Epoch 20: iteration 600, the loss is [ 0

Epoch 23: iteration 100, the loss is [ 0.1515031]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.641388879779258, auc for val: 0.643050508379444
--------------------------------------------------------------
Epoch 23: iteration 200, the loss is [ 0.13341773]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6413893762468595, auc for val: 0.6430788280263714
--------------------------------------------------------------
Epoch 23: iteration 300, the loss is [ 0.15375741]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6413701379866443, auc for val: 0.6430500714744113
--------------------------------------------------------------
Epoch 23: iteration 400, the loss is [ 0.22994156]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6413833569087315, auc for val: 0.6430591828531521
--------------------------------------------------------------
Epoch 23: iteration 500, the loss is [ 0.16

Epoch 26: iteration 0, the loss is [ 0.17229192]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.641376170988281, auc for val: 0.6431100575456526
--------------------------------------------------------------
Epoch 26: iteration 100, the loss is [ 0.20694239]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6413861193888512, auc for val: 0.6430990769486165
--------------------------------------------------------------
Epoch 26: iteration 200, the loss is [ 0.125379]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6413963914940135, auc for val: 0.6430917245014637
--------------------------------------------------------------
Epoch 26: iteration 300, the loss is [ 0.13139428]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6413878593161039, auc for val: 0.6430441372011997
--------------------------------------------------------------
Epoch 26: iteration 400, the loss is [ 0.1766

Epoch 28: iteration 1100, the loss is [ 0.15401116]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6413693516517277, auc for val: 0.6429226846717616
--------------------------------------------------------------
Epoch 29: iteration 0, the loss is [ 0.13656069]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6413722006138193, auc for val: 0.6429973586701226
--------------------------------------------------------------
Epoch 29: iteration 100, the loss is [ 0.18440996]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6413798333309989, auc for val: 0.6430208879149435
--------------------------------------------------------------
Epoch 29: iteration 200, the loss is [ 0.12946518]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6413760391354237, auc for val: 0.6429961554138675
--------------------------------------------------------------
Epoch 29: iteration 300, the loss is [ 0.

Epoch 31: iteration 1000, the loss is [ 0.13189209]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.641376368144672, auc for val: 0.6430075375676327
--------------------------------------------------------------
Epoch 31: iteration 1100, the loss is [ 0.15926169]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6413768923009718, auc for val: 0.6430015382535426
--------------------------------------------------------------
Epoch 32: iteration 0, the loss is [ 0.15928867]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.641378079700051, auc for val: 0.6430037977171098
--------------------------------------------------------------
Epoch 32: iteration 100, the loss is [ 0.14911841]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6413744900552358, auc for val: 0.6430056188617123
--------------------------------------------------------------
Epoch 32: iteration 200, the loss is [ 0.1

Epoch 34: iteration 900, the loss is [ 0.18678483]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.641378786625871, auc for val: 0.6430053997022298
--------------------------------------------------------------
Epoch 34: iteration 1000, the loss is [ 0.16108371]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6413741229087242, auc for val: 0.6430255652024783
--------------------------------------------------------------
Epoch 34: iteration 1100, the loss is [ 0.16019197]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6413773526808362, auc for val: 0.6430446221799254
--------------------------------------------------------------
Epoch 35: iteration 0, the loss is [ 0.1538405]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6413741887346859, auc for val: 0.6430252739324566
--------------------------------------------------------------
Epoch 35: iteration 100, the loss is [ 0.2

Epoch 37: iteration 800, the loss is [ 0.11982491]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6413810387739468, auc for val: 0.6430401923305159
--------------------------------------------------------------
Epoch 37: iteration 900, the loss is [ 0.16394164]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6413789165899492, auc for val: 0.6430288695619006
--------------------------------------------------------------
Epoch 37: iteration 1000, the loss is [ 0.14524147]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.641377143146914, auc for val: 0.6430212922995369
--------------------------------------------------------------
Epoch 37: iteration 1100, the loss is [ 0.17445102]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6413821933404187, auc for val: 0.6430335963370606
--------------------------------------------------------------
Epoch 38: iteration 0, the loss is [ 0.

Epoch 40: iteration 700, the loss is [ 0.19934449]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6413766717960561, auc for val: 0.6430212286725904
--------------------------------------------------------------
Epoch 40: iteration 800, the loss is [ 0.14050472]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.641379177442402, auc for val: 0.6430344178316367
--------------------------------------------------------------
Epoch 40: iteration 900, the loss is [ 0.17136577]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6413795565243907, auc for val: 0.6430299738429057
--------------------------------------------------------------
Epoch 40: iteration 1000, the loss is [ 0.18101215]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6413750464413412, auc for val: 0.6430179016902537
--------------------------------------------------------------
Epoch 40: iteration 1100, the loss is [ 

Epoch 43: iteration 600, the loss is [ 0.18498687]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6413791253201357, auc for val: 0.6430240028074585
--------------------------------------------------------------
Epoch 43: iteration 700, the loss is [ 0.1706806]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6413761624284948, auc for val: 0.643017952591811
--------------------------------------------------------------
Epoch 43: iteration 800, the loss is [ 0.14810188]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.641378178197873, auc for val: 0.6430299653593128
--------------------------------------------------------------
Epoch 43: iteration 900, the loss is [ 0.17856021]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6413799757529821, auc for val: 0.6430301477565596
--------------------------------------------------------------
Epoch 43: iteration 1000, the loss is [ 0.1

Epoch 46: iteration 500, the loss is [ 0.14536588]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6413795532692605, auc for val: 0.6430347656589444
--------------------------------------------------------------
Epoch 46: iteration 600, the loss is [ 0.13224834]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6413770639387513, auc for val: 0.6430225803917208
--------------------------------------------------------------
Epoch 46: iteration 700, the loss is [ 0.1821373]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6413804757570211, auc for val: 0.6430392237869968
--------------------------------------------------------------
Epoch 46: iteration 800, the loss is [ 0.1300853]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6413775001261681, auc for val: 0.6430218847371056
--------------------------------------------------------------
Epoch 46: iteration 900, the loss is [ 0.2

Epoch 49: iteration 400, the loss is [ 0.19317256]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6413797472107083, auc for val: 0.6430359858823852
--------------------------------------------------------------
Epoch 49: iteration 500, the loss is [ 0.14067934]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6413760116878463, auc for val: 0.6430133375172905
--------------------------------------------------------------
Epoch 49: iteration 600, the loss is [ 0.17800969]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6413777801075329, auc for val: 0.6430315546190436
--------------------------------------------------------------
Epoch 49: iteration 700, the loss is [ 0.1569825]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6413803310042041, auc for val: 0.6430373390154478
--------------------------------------------------------------
Epoch 49: iteration 800, the loss is [ 0.

Epoch 52: iteration 300, the loss is [ 0.14237863]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6413770198136559, auc for val: 0.6430148956705141
--------------------------------------------------------------
Epoch 52: iteration 400, the loss is [ 0.16675253]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6413779728031901, auc for val: 0.6430315319961293
--------------------------------------------------------------
Epoch 52: iteration 500, the loss is [ 0.15748021]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6413801491187934, auc for val: 0.6430385196487886
--------------------------------------------------------------
Epoch 52: iteration 600, the loss is [ 0.15715325]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.641380304480923, auc for val: 0.6430415002177499
--------------------------------------------------------------
Epoch 52: iteration 700, the loss is [ 0.

Epoch 55: iteration 200, the loss is [ 0.14767593]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6413828550561, auc for val: 0.6430414521440568
--------------------------------------------------------------
Epoch 55: iteration 300, the loss is [ 0.14351703]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6413783647851381, auc for val: 0.6430293135365941
--------------------------------------------------------------
Epoch 55: iteration 400, the loss is [ 0.11995388]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6413759323189364, auc for val: 0.6430093389171851
--------------------------------------------------------------
Epoch 55: iteration 500, the loss is [ 0.13109368]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6413771854636037, auc for val: 0.6430229268050962
--------------------------------------------------------------
Epoch 55: iteration 600, the loss is [ 0.16

Epoch 58: iteration 100, the loss is [ 0.17305626]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6413814694557733, auc for val: 0.6430423005033438
--------------------------------------------------------------
Epoch 58: iteration 200, the loss is [ 0.19034794]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6413845388424072, auc for val: 0.643033990824129
--------------------------------------------------------------
Epoch 58: iteration 300, the loss is [ 0.15255673]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6413827461097128, auc for val: 0.6430381491985667
--------------------------------------------------------------
Epoch 58: iteration 400, the loss is [ 0.12074572]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6413785860134161, auc for val: 0.6430270441755018
--------------------------------------------------------------
Epoch 58: iteration 500, the loss is [ 0.

Epoch 0: iteration 1000, the loss is [ 0.15908463]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6408776873761937, auc for val: 0.6426719379452104
--------------------------------------------------------------
Epoch 0: iteration 1100, the loss is [ 0.16619278]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.641040586859613, auc for val: 0.6424170370860177
--------------------------------------------------------------
Epoch 1: iteration 0, the loss is [ 0.20237914]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.525291567272532, auc for val: 0.5266543600652127
--------------------------------------------------------------
Epoch 1: iteration 100, the loss is [ 0.17061432]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6412548897079615, auc for val: 0.6425766770948216
--------------------------------------------------------------
Epoch 1: iteration 200, the loss is [ 0.216063

Epoch 3: iteration 900, the loss is [ 0.15595786]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6189265089266643, auc for val: 0.6195295152665351
--------------------------------------------------------------
Epoch 3: iteration 1000, the loss is [ 0.1621576]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.640734475119806, auc for val: 0.641826245334381
--------------------------------------------------------------
Epoch 3: iteration 1100, the loss is [ 0.12423261]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.5190007799574338, auc for val: 0.5187083314727166
--------------------------------------------------------------
Epoch 4: iteration 0, the loss is [ 0.1893075]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6410261080414779, auc for val: 0.6428672443923658
--------------------------------------------------------------
Epoch 4: iteration 100, the loss is [ 0.10812939

Epoch 6: iteration 800, the loss is [ 0.10943824]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6379576024628443, auc for val: 0.6388790550693739
--------------------------------------------------------------
Epoch 6: iteration 900, the loss is [ 0.15277018]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.575206883731994, auc for val: 0.5755451728570083
--------------------------------------------------------------
Epoch 6: iteration 1000, the loss is [ 0.14108901]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.5473930754607962, auc for val: 0.546650335505313
--------------------------------------------------------------
Epoch 6: iteration 1100, the loss is [ 0.14443119]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.639552869810307, auc for val: 0.6416468597631756
--------------------------------------------------------------
Epoch 7: iteration 0, the loss is [ 0.1549950

Epoch 9: iteration 700, the loss is [ 0.17215151]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6071677433463711, auc for val: 0.6102320644694013
--------------------------------------------------------------
Epoch 9: iteration 800, the loss is [ 0.16671154]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6207713485361435, auc for val: 0.6217488160912854
--------------------------------------------------------------
Epoch 9: iteration 900, the loss is [ 0.16707864]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6042916885266864, auc for val: 0.6076061436245295
--------------------------------------------------------------
Epoch 9: iteration 1000, the loss is [ 0.1660666]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6022933064455588, auc for val: 0.6052083267968427
--------------------------------------------------------------
Epoch 9: iteration 1100, the loss is [ 0.147

Epoch 12: iteration 600, the loss is [ 0.14273137]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.5, auc for val: 0.5
--------------------------------------------------------------
Epoch 12: iteration 700, the loss is [ 0.11679047]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.5, auc for val: 0.5
--------------------------------------------------------------
BLOWINGUP: Learning rate is 0.025221516643856644. Weight decay is 0.015984258287073554. dropout is 0.5
 Val aus is 0.5. Train auc is 0.5
Epoch 0: iteration 0, the loss is [ 1.17364573]
  acc for train: 0.03798814540029435, acc for val: 0.03827
  auc for train: 0.35931559677640257, auc for val: 0.3572875161366067
--------------------------------------------------------------
Epoch 0: iteration 100, the loss is [ 0.26455334]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.49541520372056713, auc for val: 0.49717576951609743
-------------------------

Epoch 2: iteration 800, the loss is [ 0.16032268]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6412827487177711, auc for val: 0.6429308628552868
--------------------------------------------------------------
Epoch 2: iteration 900, the loss is [ 0.16457665]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6413210135341928, auc for val: 0.642952498844966
--------------------------------------------------------------
Epoch 2: iteration 1000, the loss is [ 0.16741574]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6410162308518993, auc for val: 0.6427903787991802
--------------------------------------------------------------
Epoch 2: iteration 1100, the loss is [ 0.13771659]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6412875080794777, auc for val: 0.6427155195757082
--------------------------------------------------------------
Epoch 3: iteration 0, the loss is [ 0.16815

Process Process-1357:
Process Process-1358:


BLOWINGUP: Learning rate is 0.0055353144032044485. Weight decay is 0.020228773910792964. dropout is 0.5
 Val aus is 0.6159458420326334. Train auc is 0.6153198217782225


Traceback (most recent call last):
  File "/usr/lib/python3.5/multiprocessing/queues.py", line 342, in get
    with self._rlock:
  File "/usr/lib/python3.5/multiprocessing/process.py", line 249, in _bootstrap
    self.run()
Traceback (most recent call last):
  File "/usr/lib/python3.5/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python3.5/multiprocessing/process.py", line 249, in _bootstrap
    self.run()
  File "/usr/local/lib/python3.5/dist-packages/torch/utils/data/dataloader.py", line 34, in _worker_loop
    r = index_queue.get()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/local/lib/python3.5/dist-packages/torch/utils/data/dataloader.py", line 34, in _worker_loop
    r = index_queue.get()
  File "/usr/lib/python3.5/multiprocessing/synchronize.py", line 96, in __enter__
    return self._semlock.__enter__()
  File "/usr/lib/python3.5/mu

Epoch 0: iteration 0, the loss is [ 0.88721603]
  acc for train: 0.9635541622144714, acc for val: 0.96327
  auc for train: 0.6129201870987113, auc for val: 0.6098416438698622
--------------------------------------------------------------
Epoch 0: iteration 100, the loss is [ 0.20654897]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6410678220498842, auc for val: 0.6427860747897316
--------------------------------------------------------------
Epoch 0: iteration 200, the loss is [ 0.18505025]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6330309870743006, auc for val: 0.6340475471276239
--------------------------------------------------------------
Epoch 0: iteration 300, the loss is [ 0.16718414]
  acc for train: 0.963552482140817, acc for val: 0.96329
  auc for train: 0.6411116265349107, auc for val: 0.6426830613493931
--------------------------------------------------------------
Epoch 0: iteration 400, the loss is [ 0.13806

Process Process-1361:
Traceback (most recent call last):
KeyboardInterrupt
  File "/usr/lib/python3.5/multiprocessing/process.py", line 249, in _bootstrap
    self.run()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/local/lib/python3.5/dist-packages/torch/utils/data/dataloader.py", line 34, in _worker_loop
    r = index_queue.get()
  File "/usr/lib/python3.5/multiprocessing/queues.py", line 342, in get
    with self._rlock:
  File "/usr/lib/python3.5/multiprocessing/synchronize.py", line 96, in __enter__
    return self._semlock.__enter__()


BLOWINGUP: Learning rate is 0.020576266888179925. Weight decay is 0.0038547135436206204. dropout is 0.5
 Val aus is 0.6229894779213635. Train auc is 0.6212915788678677
This is round you consume 0:22:37.933934 time to run this model.


KeyboardInterrupt: 

ERROR:tornado.general:Uncaught exception, closing connection.
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/zmq/eventloop/zmqstream.py", line 414, in _run_callback
    callback(*args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/tornado/stack_context.py", line 277, in null_wrapper
    return fn(*args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/ipykernel/kernelbase.py", line 283, in dispatcher
    return self.dispatch_shell(stream, msg)
  File "/usr/local/lib/python3.5/dist-packages/ipykernel/kernelbase.py", line 235, in dispatch_shell
    handler(stream, idents, msg)
  File "/usr/local/lib/python3.5/dist-packages/ipykernel/kernelbase.py", line 421, in execute_request
    self._abort_queues()
  File "/usr/local/lib/python3.5/dist-packages/ipykernel/kernelbase.py", line 637, in _abort_queues
    self._abort_queue(stream)
  File "/usr/local/lib/python3.5/dist-packages/ipykernel/kernelbase.py", line 662, in _abort_queue
    poll

In [17]:
#! ls -a
# ! rm 
# X_train.describe()
for i in sorted(stack_model.train_hist, key =lambda x:x[0], reverse=True):
    print(stack_model.train_hist[i])


Learning rate is 0.0006241303598307956. Weight decay is 0.0026706282114709093. dropout is 0.5
 Val aus is 0.6431840995163449. Train auc is 0.6409540445249023
Learning rate is 0.0004130048057886514. Weight decay is 0.015377794660046963. dropout is 0.5
 Val aus is 0.6429664246626596. Train auc is 0.6408428527176809
Learning rate is 0.025221516643856644. Weight decay is 0.015984258287073554. dropout is 0.5
 Val aus is 0.6429169441072538. Train auc is 0.6408618952542339
Learning rate is 0.0055353144032044485. Weight decay is 0.020228773910792964. dropout is 0.5
 Val aus is 0.6428881663463115. Train auc is 0.6406036645498829
Learning rate is 0.020576266888179925. Weight decay is 0.0038547135436206204. dropout is 0.5
 Val aus is 0.6428809043908161. Train auc is 0.6410302614713866
Learning rate is 0.001341993635972462. Weight decay is 0.044122990686213025. dropout is 0.5
 Val aus is 0.6428361774752811. Train auc is 0.6406958356800573
Learning rate is 0.0005067940145389166. Weight decay is 0.0

In [17]:
#result.to_csv('submission04.csv', index=False)
# train_data.shape
# X_train.shape
# np.bincount(y_train)
# ! pwd
miscellany
X_test.head(10)

ps_ind_01  ps_ind_03  ps_ind_14  ps_ind_15  ps_reg_01  ps_reg_02  \
0  -0.957955   1.324746  -0.097621   1.325445  -0.385866  -0.344291   
1   1.058390   0.213594  -0.097621  -0.648589   1.004750   0.150435   
2   1.562477  -0.527174  -0.097621   0.761435  -0.733520  -1.086380   
3  -0.957955   0.583978  -0.097621  -0.930594  -1.776482  -0.591654   
4   1.562477   0.954362  -0.097621  -0.930594   1.004750  -0.096928   
5  -0.957955   0.583978  -0.097621   0.761435   1.004750   0.150435   
6  -0.957955  -0.527174  -0.097621   1.043440  -1.776482  -0.839017   
7  -0.957955  -1.638327  -0.097621  -0.084579   1.004750   1.634614   
8  -0.957955   0.954362  -0.097621  -0.366584  -0.733520  -1.086380   
9  -0.453868   0.583978  -0.097621  -0.084579   1.004750   1.387251   

      ps_reg_03  ps_car_11  ps_car_12  ps_car_13        ...         \
0 -9.076669e-01  -1.616955  -1.092966  -0.639875        ...          
1 -3.924898e-01  -1.616955  -1.092966  -0.921442        ...          
2  7.078733e-02   0.785475   0.343963   0.369450        ...          
3 -2.450736e-14  -0.415740  -0.099167  -0.717556        ...          
4 -2.440197e-01   0.785475  -0.099167  -0.001560        ...          
5 -4.290506e-01  -0.415740  -1.092966  -0.277586        ...          
6 -2.450736e-14  -0.415740  -1.092966  -0.394927        ...          
7  8.057727e-01   0.785475  -1.092966  -1.009201        ...          
8 -9.490565e-01  -0.415740   1.153809   3.681776        ...          
9 -2.450736e-14  -0.415740   1.153809   1.015509        ...          

   ps_calc_16_bin_0  ps_calc_16_bin_1  ps_calc_17_bin_0  ps_calc_17_bin_1  \
0                 0                 1                 0                 1   
1                 1                 0                 0                 1   
2                 1                 0                 1                 0   
3                 1                 0                 0                 1   
4                 0                 1                 0                 1   
5                 1                 0                 0                 1   
6                 0                 1                 1                 0   
7                 0                 1                 1                 0   
8                 1                 0                 0                 1   
9                 0                 1                 0                 1   

   ps_calc_18_bin_0  ps_calc_18_bin_1  ps_calc_19_bin_0  ps_calc_19_bin_1  \
0                 1                 0                 1                 0   
1                 0                 1                 1                 0   
2                 1                 0                 1                 0   
3                 1                 0                 1                 0   
4                 1                 0                 1                 0   
5                 1                 0                 0                 1   
6                 1                 0                 1                 0   
7                 1                 0                 1                 0   
8                 1                 0                 1                 0   
9                 1                 0                 1                 0   

   ps_calc_20_bin_0  ps_calc_20_bin_1  
0                 0                 1  
1                 0                 1  
2                 1                 0  
3                 1                 0  
4                 0                 1  
5                 1                 0  
6                 1                 0  
7                 1                 0  
8                 1                 0  
9                 1                 0  

[10 rows x 237 columns]

# Only include LB score in this chunk

#### 10.28
1. No oversample, No variable dropping: 2.75
2. 0.2 SMOTE oversample, No variable dropping: 2.76
3. Take fold outside(ensemble), only fold one, 0.2 SMOTE oversample, No variable dropping: 2.76 (no improvement)
---
#### 10.29
4. 0.4 SMOTE oversample, No variable dropping: 2.7
5. META: Logistic Version, 0.3: 2.56(Don't know what is wrong here)
6. 0.1 SMOTE oversample, No variable dropping: 2.75
7. 0.15 SMOTE oversample, add L2:0.034, No variable dropping: 2.75

#### 10.30
###### (correct the mistake in stacker, forget to take mean for S_test) 
8. 0.22 SMOTE oversample, no META: 0.278
9. 0.20 SMOTE oversample, no META: 0.279 Stacker score: 0.94613
10. 0.25 SMOTE oversample, no META: 0.279 Stacker score: ***
11. 0.18 SMOTE oversample, no META: 0.2 Stacker score: 0.9
###### (won't normalize binary and categorical anymore  
12. 0.18 SMOTE oversample, no META: 0.2 Stacker score: 0.9

###### () 
9. 

# Only include offline test score(auc) in this chunk

#### 10.29

all based on rso66 with 0.1 test set splitted from train_data,ensemble: three lgb in one.
0. As I rememberred, 0.2 SMOTE oversample, No variable dropping: 0.6424
1. 0.4 SMOTE oversample, No variable dropping: 0.641637
2. 0.6 SMOTE oversample, No variable dropping: 0.639938
3. 0.1 SMOTE oversample, No variable dropping: 0.6431368
4. 0.0 SMOTE oversample, No variable dropping: 0.4736901 #what's wrong with it here??
5. 0.3 SMOTE oversample, No variable dropping: 0.641200
6. META: Logistic Version: (0.1, 0.2, 0.3, 0.5, 0.7, 1)
        (0.633452, 0.632539, 0.645008, 0.628007, 0.627307, 0.625253)

In [ ]:
import data_util
import data_preprocess
from sklearn.model_selection import train_test_split
from datetime import datetime
import numpy as np 
from lgb_stacker import lgb_ensemble
from Stacker import Ensemble
from sklearn import metrics
#from sklearn.neighbors import KNeighborsClassifier

rso = np.random.RandomState(66)

result = []
for os in [None]:
    train_data = data_util.load_train_data()
    test_data= data_util.load_test_data()
    
    lv1_pre = data_preprocess.preprocess_cell()
    
    X_train = train_data.drop(['id','target'],axis=1)
    y_train = train_data['target']
    
    X_train, X_train_test, y_train, y_train_test = train_test_split(X_train,y_train,test_size =0.1 ,random_state=rso)

    X_train_af, y_train_af, col = lv1_pre.process(X_train, y=y_train, oversample=os, rso =rso)
    X_train_test_af = lv1_pre.process(X_train_test,test=True,rso =rso)
    
    print(type(X_train_af), type(y_train_af), type(X_train_test_af))
    stack_model = Ensemble(n_splits=5)
    y_pred= stack_model.fit_predict(X_train_af, y_train_af, X_train_test_af)

    result.append(metrics.roc_auc_score(y_train_test,y_pred))

In [11]:
from sklearn import metrics
metrics.roc_auc_score(y_train_test,y_pred)
result

[0.64313174736945844, 0.59310226899327134]

# change to the standard lgb

In [ ]:
from Stacker import Ensemble

tic= datetime.now()
stack_model = Ensemble(n_splits=5)
y_pred= stack_model.fit_predict(X_train,y_train, X_test)
toc = datetime.now()
print('This is round you consume {} time to run this model.'.format(toc-tic))

result['target'] = y_pred
result.to_csv('submission08.csv', index=False)
stack_model.miscellany